editor/plugins/spatial_editor_plugin.cpp

```diff
@@ -2552,6 +2557,14 @@ void SpatialEditorViewport::_menu_option(int p_option) {
 			call_deferred("update_transform_gizmo_view");
 			_update_name();
 
+		} break;
+		case VIEW_LOCK_ROTATION: {
+
+			int idx = view_menu->get_popup()->get_item_index(VIEW_LOCK_ROTATION);
+			bool current = view_menu->get_popup()->is_item_checked(idx);
+			lock_rotation = !current;
```

Comment:
I would also change the top menu (that shows [Top],[Bottom], etc and add a lock icon, you could try doing something like
view_menu->set_icon( get_icon("Lock", "EditorIcons") )
and to clear it
view_menu->set_icon( Ref<Texture() )

Reply:
Done